app/src/main/java/com/fastlink/zekrapp/presentation/ZekirScreen.kt

```diff
@@ -0,0 +1,211 @@
+                ) {
+                    Text(
+                        text = zekirCounter.intValue.toString(),
+                        style = MaterialTheme.typography.titleLarge,
+                        color = MaterialTheme.colorScheme.onPrimaryContainer
+                    )
+                }
+            }
+        },
+        isFloatingActionButtonDocked = true,
+        floatingActionButtonPosition = FabPosition.Center,
+        bottomBar = {
+            BottomBar(bottomBarItems = getListOfBottomBarItemsInZekirScreen(
+                zekirCategory = category,
+                onCopyIconClicked = {
+                    clipboardManager.setText(AnnotatedString(viewModel.zekirs[zekirNumber.intValue].zekirTitle))
+                },
+                onFavoriteIconClicked = {
+                    viewModel.updateCategory(
+                        categoryId = category.id,
+                        isFavorite = !category.isFavorite
+                    )
+                }
+            ))
+        }
+    ) {
+        Column(
+            horizontalAlignment = Alignment.End,
+            modifier = Modifier
+                .padding(it)
+                .fillMaxSize()
+                .verticalScroll(rememberScrollState())
+                .clickable(
+                    enabled = buttonIsEnable(),
+                    onClick = {
+                        Log.d(
+                            "ZekirScreen",
+                            "onClick: ${viewModel.zekirs.size} && ${zekirNumber.intValue}"
+                        )
+                        viewModel.onClickedZekirCard(scope)
+                    }
+                )
+                .padding(
+                    start = 16.dp,
+                    end = 16.dp,
+                    bottom = 60.dp,
+                    top = 20.dp
+                )
+                .clip(RoundedCornerShape(10.dp))
+                .background(MaterialTheme.colorScheme.primaryContainer.copy(alpha = 0.5f))
+                .padding(12.dp)
+                .dashedBorder(
+                    brush = Brush.linearGradient(
+                        colors = listOf(
+                            MaterialTheme.colorScheme.primary,
+                            MaterialTheme.colorScheme.primary.copy(alpha = 0.7f)
+                        )
+                    ),
+                    shape = RoundedCornerShape(4.dp)
+                )
+                .padding(16.dp),
+        ) {
+            Text(
+                text = viewModel.zekirs[zekirNumber.intValue].zekirTitle,
+                modifier = Modifier.fillMaxWidth(),
+                textAlign = TextAlign.End,
+                color = MaterialTheme.colorScheme.onSurface,
+                style = MaterialTheme.typography.titleMedium,
+            )
+            Text(
+                text = "\n(${
+                    if (viewModel.zekirs[zekirNumber.intValue].textCounter == "null") stringResource(
```

Comment:
Why would you have a null value as a string?

Reply:
I explained that we have two fields or instances: `textCounter `and `numericCounter`. In the Zekir database, we have a column with two types of data: if Zekir is read once, the data is `null`, but if it is read more than once, the data is a `String`. I store this data in an ArrayList of type `Any`, which allows you to save either `null `or a `String`. However, this gives a warning because Any doesn't match the type `String?.` To resolve this, I store the data as a String by calling the `.toString` function when saving the data in the model. This way, I can check if the data is a null string, and if it is, I use a String Resource; if it's not null, I use `textCounter`

Reply:
I don't really want the explanation, I'm only pointing out to this. Having it empty is better, null makes sense only as a value, not in a string.

Reply:
you are right, if I checked in first maybe is better 
![image](https://github.com/user-attachments/assets/855b9835-9933-4e7d-b9cd-d1bc16c603d9)
in the singleton file 
instead of the previous code I change it to this 
`if(i[5].toString()=="null") null else i[5].toString()`
and in the ZekirScreen Refactor the code to 
`viewModel.zekirs[zekirNumber].textCounter ?: stringResource(the text)`

this is better or not ????????????????????????????????????????????????????????????????

Reply:
> you are right, if I checked in first maybe is better ![image](https://private-user-images.githubusercontent.com/121394643/359825134-855b9835-9933-4e7d-b9cd-d1bc16c603d9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQyMzA4NzksIm5iZiI6MTcyNDIzMDU3OSwicGF0aCI6Ii8xMjEzOTQ2NDMvMzU5ODI1MTM0LTg1NWI5ODM1LTk5MzMtNGU3ZC1iOWNkLWQxYmMxNmM2MDNkOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMVQwODU2MTlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2EwN2YwZTQ2ZmViZDJmOGIyZTVkMzA2MjhjOTdjYTlmMjQzMjFhNzE3NzM3YWNlYjVjMWVjODA4YTEwZDAzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.fK7z_IyrP2XaA5O9TMtFuECF_-9oOsC41llhx3Miayc) in the singleton file instead of the previous code I change it to this `if(i[5].toString()=="null") null else i[5].toString()` and in the ZekirScreen Refactor the code to `viewModel.zekirs[zekirNumber].textCounter ?: stringResource(the text)`
> 
> What do you think, is this better or not?

Reply:
You are still checking null inside a string, so nothing's changed.

Reply:
the data like this 
![image](https://github.com/user-attachments/assets/0c8f387c-3c32-4576-92f8-c577ab086c4b)
![image](https://github.com/user-attachments/assets/7caa9b1a-e6e8-4118-bd09-d509f7d8d106)
What is your suggestion about this? instead of checking on how to handle this?

Reply:
First of all I don't know why the type is Any instead of having a data class representing the data. You can have your data like this:

val zekirData: ArrayList<ZekirModel> = arrayListOf(

Reply:
you are right I didn't thought that I change it to that inshallah thank you